doc/i2p.md

```diff
@@ -47,9 +53,34 @@ In a typical situation, this suffices:
 bitcoind -i2psam=127.0.0.1:7656
 ```
 
-The first time Bitcoin Core connects to the I2P router, its I2P address (and
-corresponding private key) will be automatically generated and saved in a file
-named `i2p_private_key` in the Bitcoin Core data directory.
+The first time Bitcoin Core connects to the I2P router, if a persistent I2P
+address is to be used, then it (and its corresponding private key) will be
+automatically generated and saved in a file named `i2p_private_key` in the
+Bitcoin Core data directory. A persistent I2P address is used for accepting
+incoming connections (`-i2pacceptincoming`) and for making outgoing connections
+if `-i2ptransientout` is not used.
+
+## Persistent vs transient I2P addresses
+
+In I2P connections, the host that receives the connection knows the I2P address
+of the connection initiator. This is unlike the Tor network where the recipient
+does not know who is connecting to them, not even the initiator's Tor address.
+
+So, even if an I2P node is not accepting incoming connections, they are known
+to other nodes by their outgoing I2P address. This creates an opportunity to
+white-list given nodes or treat them differently based on their I2P address.
+However, this also creates an opportunity to fingerprint or analyze a given node
+because it always uses the same I2P address when it connects to other nodes. If
+this is undesirable, then a node operator can use `-i2ptransientout` to generate
```

Comment:
Oh, well, it is also possible to analyze a node by connecting to it. That is kind of unavoidable and a node that wants to avoid this should not be listening.